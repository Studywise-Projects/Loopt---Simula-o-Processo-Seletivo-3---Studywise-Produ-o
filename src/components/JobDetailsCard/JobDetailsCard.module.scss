@import '@/styles/_global.scss';

.card {
  display: none;

  @media only screen and (min-width: 1024px) {
    display: flex;
    border-radius: 20px;
    box-shadow: 4px 4px 15px 0px rgba(0, 0, 0, 0.10);
    flex-direction: column;
    width: 35%;
  }
}

.cardContent {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.jobDetailsContainer {
  display: flex;
  flex-direction: column;
  gap: 8px;
}