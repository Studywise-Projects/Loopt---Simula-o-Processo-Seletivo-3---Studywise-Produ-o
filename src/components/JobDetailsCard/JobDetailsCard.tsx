import { Box, Card, CardContent } from '@mui/material';
import SearchBar from '../SearchBar/SearchBar';
import { Typography } from '..';
import { IJob } from '@/interfaces/IJob';
import styles from './JobDetailsCard.module.scss';
import { ChangeEventHandler } from 'react';

interface IJobDetailsCard {
  job: IJob;
  value: string;
  handleChange: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
}

function JobDetailsCard({ job, value, handleChange }: IJobDetailsCard) {
  return (
    <Card className={styles.card}>
      <CardContent className={styles.cardContent}>
        <SearchBar
          id='searchbar-candidates'
          label='Buscar candidato pelo nome'
          value={value}
          handleChange={handleChange}
          isAuxSearch={true}
        />
        <Typography variant='subtitle' text='Detalhes da Vaga' />
        <Box className={styles.jobDetailsContainer}>
          <Typography variant='body' text='Requisitos' />
          {job?.requirements.map((requirement: string, index: number) => (
            <Typography variant='caption' text={requirement} key={index} />
          ))}
          <Typography variant='body' text='Diferenciais' />
          {job?.differentials.map((differential: string, index: number) => (
            <Typography variant='caption' text={differential} key={index} />
          ))}
        </Box>
      </CardContent>
    </Card>
  );
}

export default JobDetailsCard;
