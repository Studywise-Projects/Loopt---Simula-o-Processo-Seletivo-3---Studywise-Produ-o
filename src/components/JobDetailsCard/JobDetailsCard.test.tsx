import { fireEvent, render } from '@testing-library/react';
import JobDetailsCard from './JobDetailsCard';
import { ChangeEvent } from 'react';

describe('JobDetailsCard', () => {
  it('Should match the snapshot', () => {
    const { getByText } = render(
      <JobDetailsCard
        job={{
          id: 1,
          label: 'Desenvolvedor Front-End',
          requirements: ['ReactJs', 'NextJs', 'React Native'],
          differentials: ['Python', 'Figma'],
        }}
        value={''}
        handleChange={() => undefined}
      />,
    );

    const element = getByText('ReactJs');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByText } = render(
      <JobDetailsCard
        job={{
          id: 1,
          label: 'Desenvolvedor Front-End',
          requirements: ['ReactJs', 'NextJs', 'React Native'],
          differentials: ['Python', 'Figma'],
        }}
        value={''}
        handleChange={() => undefined}
      />,
    );

    const element = getByText('Figma');
    expect(element).toBeInTheDocument();
  });

  it('Should call handleChangeMock function once', () => {
    const handleChangeMock = jest.fn();

    const { getByLabelText } = render(
      <JobDetailsCard
        job={{
          id: 1,
          label: 'Desenvolvedor Front-End',
          requirements: ['ReactJs', 'NextJs', 'React Native'],
          differentials: ['Python', 'Figma'],
        }}
        value={''}
        handleChange={handleChangeMock}
      />,
    );

    const element = getByLabelText('Buscar candidato pelo nome');
    fireEvent.change(element, { target: { value: 'value changed' } });
    expect(handleChangeMock).toHaveBeenCalledTimes(1);
  });
});
