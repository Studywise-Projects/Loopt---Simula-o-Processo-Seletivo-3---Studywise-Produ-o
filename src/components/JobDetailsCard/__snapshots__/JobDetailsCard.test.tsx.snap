// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetailsCard Should match the snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 undefined css-ahj2mt-MuiTypography-root"
>
  ReactJs
</p>
`;
